packages/node_modules/@ciscospark/widget-recents/src/components/RecentsHeader.js

```diff
@@ -2,56 +2,110 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
 
-import {Button, Icon, SearchInput} from '@collab-ui/react';
+import {Button, Icon, Popover, SearchInput} from '@collab-ui/react';
 
 import PresenceAvatar from '@ciscospark/react-container-presence-avatar';
 
-
+import PopoverContent from './PopoverContent';
 import styles from './RecentsHeader.css';
 
 const propTypes = {
   currentUser: PropTypes.shape({
     id: PropTypes.string,
-    displayName: PropTypes.string
+    displayName: PropTypes.string,
+    email: PropTypes.string,
+    img: PropTypes.string
   }).isRequired,
   enableAddButton: PropTypes.bool.isRequired,
   enableSpaceListFilter: PropTypes.bool.isRequired,
   enableUserProfile: PropTypes.bool.isRequired,
+  enableUserProfilePopover: PropTypes.bool.isRequired,
   hideBottomBorder: PropTypes.bool.isRequired,
   onAddClick: PropTypes.func,
   onFilterChange: PropTypes.func,
-  onProfileClick: PropTypes.func
+  onProfileClick: PropTypes.func,
+  onSignOutClick: PropTypes.func
 };
 
 const defaultProps = {
   onAddClick: () => {},
   onProfileClick: () => {},
-  onFilterChange: () => {}
+  onFilterChange: () => {},
+  onSignOutClick: () => {}
 };
 
+/**
+ * Displays the profile avatar if needed.
+ * Displays the popover content if needed
+ *
+ * @param {bool} enableUserProfile to display the profile avatar
+ * @param {bool} enableUserProfilePopover to display the profile popover content
+ * @param {func} onProfileClick to notify if the avtar had been clicked
+ * @param {func} onSignOutClick to notify if the sign out link had been clicked
+ * @param {obj} currentUser current user information
+ * @returns {obj} JSX element with or without avatar profile
+ */
+function displayProfileAvatar(
```

Comment:
Can you use [property destructuring](https://github.com/webex/web-styleguide/tree/master/javascript#destructuring--object) here? Not sure why eslint didn't catch this one.

Reply:
property destruction occurs here:
```js
const {
    currentUser,
    enableAddButton,
    enableSpaceListFilter,
    enableUserProfile,
    enableUserProfilePopover,
    hideBottomBorder,
    onAddClick,
    onFilterChange,
    onProfileClick,
    onSignOutClick
  } = props;
```
I could not send the props to our helper method, eslint complains that the props have been passed to helper method are being declared but not used.

Reply:
Therefore, I destruct the property first and pass the items one by one:
```js
 {displayProfileAvatar(
          enableUserProfile,
          enableUserProfilePopover,
          onProfileClick,
          onSignOutClick,
          currentUser
        )}
```

Reply:
I meant the property destructuring in the props of the function like in the example:

```js
// best
function getFullName({ firstName, lastName }) {
  return `${firstName} ${lastName}`;
}
```

Reply:
not related anymore